app/scripts/modules/core/src/search/widgets/Search.tsx

```diff
@@ -256,7 +276,16 @@ export class Search extends React.Component<ISearchProps, ISearchState> {
     this.inputElement.focus();
   }
 
-  private handleFilterChange(key: Key): void {
+  private filterClicked(filter: Filter): void {
+    this.setState({
+      activeFilter: filter.props.filterType
+    }, () => {
```

Comment:
`setState` callbacks are discouraged. Is there a reason you are using a callback instead of just putting this in `componentDidUpdate()`?

It is also odd that after this setState resolves setting `activeFilter`, you (may) immediately call another `setState` inside `handleFilterSelection()` to set `activeFilter` again. Is there benefit in doing these so quickly in succession or can this be refactored?

Reply:
Good catch.